### DISCUSSION TITLE
[Feature request] Distinct state time and percent in state on stat panel

### DISCUSSION THREAD

**Discussion by TenGbps** [grafana/grafana#59061] [Dashboards, Panels & Visualizations]
Hello,

Could you add a discrete state time calculation on a stat panel to get a state elapsed time?
The period is obtained from the Timepicker and the state can be set with a "Value mappings".
In this case we using elasticsearch data source.

Here is an example of the value sought in the legend (percent, time and count state change) with the discrete panel plugins:
![Capture d’écran 2022-11-21 à 17 26 21](https://user-images.githubusercontent.com/30792994/203108158-c2b5f422-8e4e-4763-af64-17d3bae32acf.png)

The stat panel could look like this showing the elapsed time in the mapped Warning state:
![Capture d’écran 2022-11-21 à 17 28 38](https://user-images.githubusercontent.com/30792994/203108212-9fb5cfba-cae7-469a-8722-619beeb1fcc1.png)

I hope to have been clear...
Kind Regards

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
